osmapi/OsmApi.py

```diff
@@ -1912,28 +1925,25 @@ def _http_request(self, cmd, path, auth, send):  # noqa
                     ).strip()
                 base64_user_pass = base64_user_pass.decode('utf-8')
 
-            self._conn.putheader(
-                'Authorization',
-                'Basic ' + base64_user_pass
-            )
+            headers['Authorization'] = 'Basic ' + base64_user_pass
         if send is not None:
-            self._conn.putheader('Content-Length', len(send))
-        self._conn.endheaders()
-        if send:
-            self._conn.send(send)
-        response = self._conn.getresponse()
-        if response.status != 200:
-            payload = response.read().strip()
-            if response.status == 410:
+            headers['Content-Length'] = len(send)
+        if send:  # If we have additional data to send
+            response = method(path, headers=headers, data=send)
+        else:
+            response = method(path, headers=headers)
```

Comment:
There is not need for this condition, if send is `None`, it will be ignored, always use `response = method(path, headers=headers, data=send)`